### PR TITLE
scripts: fix a typo while to check build_type

### PR DESCRIPTION
**Pull request by Kvasscn**
check $build_type is not an empty string instead of equal to "true".

Fixes: #3635

Signed-off-by: zhanghj <zhanghj.lc@inspur.com>

### CODE REVIEW

**Comment by liubin** [comment]
@jcvenegas PTAL, refer to [this commit](https://github.com/kata-containers/kata-containers/commit/ff6a677d167ca459cc0232f9b2901bcbb15548c5).

**Review by liubin — APPROVED** [review]

File: tools/packaging/kernel/build-kernel.sh [hunk 1]
```diff
@@ -352,7 +352,7 @@ setup_kernel() {
 	${packaging_scripts_dir}/apply_patches.sh "${patches_dir_for_version}"
 
 	# Apply version specific patches for build_type build
-	if [ "${build_type}" == "true" ] ;then
+	if [ "${build_type}" != "" ] ;then
```
Comment (jodh-intel) [inline tools/packaging/kernel/build-kernel.sh:355]: This could be simplified slightly to:

```suggestion
	if [ -n "${build_type}" ] ; then
```

**Review by jodh-intel — APPROVED** [review]
Thanks @Kvasscn.

lgtm

**Pull request merged**